uc-catalog/src/lib.rs

```diff
@@ -63,8 +66,10 @@ impl<'a> UCCatalog<'a> {
             start_version: Some(0),
             end_version: version.and_then(|v| v.try_into().ok()),
         };
-        // TODO: does it paginate?
-        let commits = self.client.get_commits(req).await?;
+        let mut commits = self.client.get_commits(req).await?;
+        if let Some(commits) = &mut commits.commits {
+            commits.sort_by_key(|c| c.version);
+        }
 
         // if commits are present, we ensure they are sorted+contiguous
         if let Some(commits) = &commits.commits {
```

Comment:
```suggestion
        // if commits are present, we ensure they are sorted+contiguous
        if let Some(commits) = &mut commits.commits {
            commits.sort_by_key(|c| c.version);
```

That suggestion isn't very readable, just saying we can collapse the two checks

Reply:
(swapping to as_mut tho)